shor/layers.py

```diff
@@ -16,7 +16,7 @@ class Qbits(_Layer, Iterable):
     def __init__(self, num, state=0, **kwargs):
         self.num = num
         self.state = state
-        self._qbits = list(range(num))
+        self._qbits = [(i, self) for i in range(num)]
```

Comment:
Probably also want to override the getter on the qbit class...

So this should be true...
qbits[0] = qbits[(0, qbits)] = (0, qbits)

See  __getitem__

Can check if key is a tuple, or an int,